packages/bsky/src/subscription/repo.ts

```diff
@@ -123,18 +98,45 @@ export class RepoSubscription {
     await this.run()
   }
 
-  private async handleMessage(msg: ProcessableMessage) {
-    if (message.isCommit(msg)) {
-      await this.handleCommit(msg)
-    } else if (message.isHandle(msg)) {
-      await this.handleUpdateHandle(msg)
-    } else if (message.isTombstone(msg)) {
-      await this.handleTombstone(msg)
-    } else if (message.isMigrate(msg)) {
-      // Ignore migrations
-    } else {
-      const exhaustiveCheck: never = msg
-      throw new Error(`Unhandled message type: ${exhaustiveCheck['$type']}`)
+  private async handleMessage(seq: number, msg: ProcessableMessage) {
+    const item = this.consecutive.push(seq)
```

Comment:
One material change in this PR is to track the consecutive item during `handleMessage()` (i.e. _during_ queue processing) rather than ahead of the item on the queue being processed.  This might actually be something to revisit, since items can begin processing out-of-order, which I believe could cause reordering of the consecutive items.